src/vm-vcpu/src/vcpu/mod.rs

```diff
@@ -52,17 +53,20 @@ pub enum Error {
 /// Dedicated Result type.
 pub type Result<T> = result::Result<T, Error>;
 
-pub(crate) struct VcpuState {
+pub struct VcpuState {
     pub id: u8,
     pub cpuid: CpuId,
     pub kernel_load_addr: GuestAddress,
+    // todo: make the load_addr and zero_page_start addr uniform.
+    // they should either both be u64 or GuestAddress.
```

Comment:
I vote for `GuestAddress`.

Reply:
Not only are they linux specific, but they're also a bit linux-loader specific (because we do not have a firmware that sets the registers up). We're probably better off having something like passing the registers directly (which is also why I was a bit on the fence weather to use GuestAddress or u64, since the registers are using u64 for their definition).

Reply:
Updated zero_page_start to GuestAddress.